src/main/kotlin/org/rust/lang/core/stubs/StubInterfaces.kt

```diff
@@ -6,16 +6,58 @@
 package org.rust.lang.core.stubs
 
 import com.intellij.util.BitUtil
-import org.rust.lang.core.psi.ext.QueryAttributes
-import org.rust.lang.core.psi.ext.RsDocAndAttributeOwner
-import org.rust.lang.core.psi.ext.getTraversedRawAttributes
-import org.rust.lang.core.psi.ext.name
+import org.rust.lang.core.psi.ext.*
+import org.rust.lang.core.resolve.KNOWN_DERIVABLE_TRAITS
 import org.rust.stdext.BitFlagsBuilder
+import org.rust.stdext.HashCode
 
 interface RsNamedStub {
     val name: String?
 }
 
+/**
+ * A common interface for stub for elements that can hold attribute or derive procedural macro attributes
+ */
+interface RsAttrProcMacroOwnerStub : RsAttributeOwnerStub {
+    /** Non-null if [mayHaveCustomDerive] is `true` */
```

Comment:
The same about comment

Reply:
This interface is a kind of backport from my ongoing attribute macro implementation, so I made an exactly reverse thing - I adopted these comments to derive-only implementation. So I can't forget it because it's already written:
![image](https://user-images.githubusercontent.com/3221931/113917921-838b0680-97ea-11eb-8dfa-7c2b4773056e.png)